File: ctcode.cs
public class LoggerImplementation : S84.CTCode.System.ctcode.OutputStream
{
    public void WriteLine(string? line)
    {
        Console.WriteLine(line ?? "");
    }
}
public class FileImplementation : S84.CTCode.System.ctcode.OutputStream
{
    public FileImplementation(string? file_name)
    {
        try
        {
            Directory.GetParent(file_name ?? "").Create();
        }
        catch { }
        this.destination = new System.IO.StreamWriter(file_name??"");
    }
    public void WriteLine(string? line)
    {
        this.destination.WriteLine(line ?? "");
        this.destination.Flush();
    }

    private System.IO.StreamWriter destination;
}
public class SystemImplementation : S84.CTCode.System.ctcode.System
{
    public string? ReadFileToString(string? file_name)
    {
        return System.IO.File.ReadAllText(file_name ?? "");
    }
    public S84.CTCode.System.ctcode.OutputStream? OpenFileWriter(string? file_name)
    {
        return new FileImplementation(file_name);
    }
    public S84.CTCode.System.ctcode.OutputStream? GetLoggerDestination()
    {
        return new LoggerImplementation();
    }
}
public class CTCodeMain
{
    public static int Main(string[] args)
    {
        S84.CTCode.Main.ctcode.Main main = new S84.CTCode.Main.ctcode.Main();
        SystemImplementation system = new SystemImplementation();
        string c_t_code_file_name = "";
        string transpiler_name = "";

        if (args.Length == 2) {
            c_t_code_file_name = args[0];
            transpiler_name = args[1];
        }

        return main.RunMain(system, c_t_code_file_name, transpiler_name) ?? 0;
    }
}

File: ctcode.rb
require 'S84.CTCode.System.ctcode'
require 'S84.CTCode.Main.ctcode'

require 'fileutils'

class FileImplementation < ::S84::CTCODE::SYSTEM::CTCODE::OutputStream
    def initialize(file_name)
        directory = File.dirname(file_name)
        FileUtils.mkdir_p(directory)
        @file = File.open(file_name, 'w')
    end

    def WriteLine(line)
        @file.puts line
        @file.flush
    end
end

class LoggerImplementation < ::S84::CTCODE::SYSTEM::CTCODE::OutputStream
    def initialize()
    end

    def WriteLine(line)
        puts line
    end
end

class SystemImplementation < ::S84::CTCODE::SYSTEM::CTCODE::System
    def initialize()
    end

    def ReadFileToString(file_name)
        return File.read(file_name);
    end

    def OpenFileWriter(file_name)
        return FileImplementation.new(file_name)
    end

    def GetLoggerDestination()
        return LoggerImplementation.new()
    end
end

def main(argv)
    system = SystemImplementation.new()
    main = ::S84::CTCODE::MAIN::CTCODE::Main.new()
    c_t_code_file_name = "";
    transpiler_name = "";

    if (argv.length == 2)
        c_t_code_file_name = argv[0];
        transpiler_name = argv[1];
    end

    logger = system.GetLoggerDestination();
    return main.RunMain(system, c_t_code_file_name, transpiler_name);
end

exit(main(ARGV))